clang/lib/Sema/SemaDecl.cpp

```diff
@@ -11868,8 +11868,10 @@ static bool CheckMultiVersionFunction(Sema &S, FunctionDecl *NewFD,
     return false;
 
   if (!OldDecl || !OldDecl->getAsFunction() ||
-      OldDecl->getDeclContext()->getRedeclContext() !=
-          NewFD->getDeclContext()->getRedeclContext()) {
+      (OldDecl->getDeclContext()->getRedeclContext() !=
+           NewFD->getDeclContext()->getRedeclContext() &&
+       OldDecl->getDeclContext()->getEnclosingNamespaceContext() !=
+           NewFD->getDeclContext()->getEnclosingNamespaceContext())) {
```

Comment:
```suggestion
      !OldDecl->getDeclContext()->getRedeclContext()->Equals(
          NewFD->getDeclContext()->getRedeclContext())) {
```

Reply:
Thanks for the suggestion! I'll push a revised version.